ui/src/db-tables/FoldersTable.ts

```diff
@@ -50,19 +31,19 @@ export class FoldersTable {
       createTime: Date.now(),
       type: "folder",
     };
-    this.records[folder.id] = folder;
-    indexdb.folders.add(folder);
-    saveDB("folders", JSON.stringify(this.records));
-    updateWorkspaceIndexDB();
-
+    await indexdb.folders.add(folder);
+    await this.listAll().then((all) => {
+      saveDB("folders", JSON.stringify(all));
+    });
```

Comment:
here should do:
```
    const records = await this.getRecords();
    records[md.id] = md;
    // save disk file db
    saveDB("media", JSON.stringify(records));
```
because db type is {[id]: Folder} but listAll() will return Folder[] it's different

Reply:
done